From f1be5f16c6abb1d9288488cd497a66544f781493 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Thu, 18 Apr 2013 18:32:00 +0200
Subject: [PORT FROM MAIN]wifi: Report error when failing to get netmask property

BZ: 94520

Exiting gracefully (with -1) from libnetutils
in case the netmask prop is not correcly retrieved.

Change-Id: I525ecee72d7097e7ea5d984daa91089baa460662
Orig-Change-Id: I545fc76e2c639e20637b16d270649497e1620a5b
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Paul Naoumenko <paulx.naoumenko@intel.com>
---
 libnetutils/dhcp_utils.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/libnetutils/dhcp_utils.c b/libnetutils/dhcp_utils.c
index e1df874..de73012 100644
--- a/libnetutils/dhcp_utils.c
+++ b/libnetutils/dhcp_utils.c
@@ -139,6 +139,9 @@ static int fill_ip_info(const char *interface,
             mask = mask << 1;
         }
         *prefixLength = p;
+    } else {
+        snprintf(errmsg, sizeof(errmsg), "Could not get the value of property:  %s", prop_name);
+        return -1;
     }
 
     for (x=0; dns[x] != NULL; x++) {
-- 
1.7.4.1

